WordPress/Classes/Utility/PushNotificationsManager.swift

```diff
@@ -339,63 +316,59 @@ final public class PushNotificationsManager: NSObject {
 
         return true
     }
+}
 
 
-
-
-    // MARK: - Private Methods: Helpers
-
+// MARK: - Private Methods
+//
+private extension PushNotificationsManager {
 
     /// Tracks a Notification Event
     ///
     /// - Parameter userInfo: The Notification's Payload
     ///
-    fileprivate func trackNotificationWithUserInfo(_ userInfo: NSDictionary) {
+    func trackNotification(with userInfo: NSDictionary) {
         var properties = [String: String]()
 
-        if let noteId = userInfo.number(forKey: notificationIdentifierKey) {
-            properties[trackingIdentifierKey] = noteId.stringValue
+        if let noteId = userInfo.number(forKey: Constants.notificationIdentifierKey) {
+            properties[Tracking.identifierKey] = noteId.stringValue
         }
 
-        if let type = userInfo.string(forKey: notificationTypeKey) {
-            properties[trackingTypeKey] = type
+        if let type = userInfo.string(forKey: Constants.notificationTypeKey) {
+            properties[Tracking.typeKey] = type
         }
 
         if let theToken = deviceToken {
-            properties[trackingTokenKey] = theToken
+            properties[Tracking.tokenKey] = theToken
         }
 
         let event: WPAnalyticsStat = (applicationState == .background) ? .pushNotificationReceived : .pushNotificationAlertPressed
         WPAnalytics.track(event, withProperties: properties)
     }
+}
 
 
+// MARK: - Nested Types
+//
+private extension PushNotificationsManager {
 
-    /// Parses the NSData sent by Apple's Push Service, and extracts the Device Token
-    ///
-    fileprivate func parseTokenFromAppleData(_ tokenData: Data) -> String {
-        return tokenData.hexString
+    enum Constants {
+        static let deviceTokenKey = "apnsDeviceToken"
+        static let deviceIdKey = "notification_device_id"
+        static let notificationBadgePath = "aps.badge"
+        static let notificationIdentifierKey = "note_id"
+        static let notificationTypeKey = "type"
+        static let notificationOriginKey = "origin"
+        static let notificationBadgeResetValue = "badge-reset"
     }
 
+    enum Helpshift {
```

Comment:
I love the `Constants` enums—I use them all the time now 😀

Here, both `Helpshift` and `Tracking` are really still collections of Constants. What do you think of putting them within `Constants`? Their values would be `Constants.Helpshift.originValue` and `Constants.Tracking.typeKey`, etc. Also makes me wonder if `Constants.Notifications.*` would make sense here, as well.

Reply:
Thanks for the review sir! but i think it's an overkill, because:

- Everything is already namespaced. It's actually **PushNotificationsManager.Helpshift** and **PushNotificationsManager.Tracking**, so there won't be a conflict.
- They're constants because of the let keyword + we're not mixing them with other `static var`'s

Thanks for the suggestion though!!

Reply:
I still don't know what the difference between the values in `Constants` and the others is, then. There are now three enums of constants, that were previously in four groups all called constants. It would make sense to call them `DeviceKeys`, `NotificationKeys`, `Helpshift`, and `Tracking` to match the previous `MARK:` groups, but I don't understand why the first two have been combined into just `Constants`.

Reply:
Actually, no difference, and you're right there. I usually use **Constants** as a "Bag of not necessarily related constants".

I'll split `Constants` into two enums, for consistency. Thanks for bringing this up!